src/js/legend.js

```diff
@@ -0,0 +1,228 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
+        strings: {
+            legendColHeading:"Legend",
+            labelColHeading:"Label",
+            valueColHeading:"Value"
+        },
+        model: {
+            // dataSet accepts:
+            // 1. an array of objects. Must contain "id", "value" and "label" variables.
+            // Example: [{id: string, value: number, label: string} ... ]
+            dataSet: []
+            // "dataSetWithColors" consolidates the dataset in original order with the array of colors, into a single array of objects
```

Comment:
We feel colors are more belong to user configuration options instead of looking up in the data set whenever there's a color change.